Sources/ContainerClient/Parser.swift

```diff
@@ -317,10 +317,30 @@ public struct Parser {
                 let s = "mode=\(val)"
                 fs.options.append(s)
             case "source":
-                let absPath = URL(filePath: val).absoluteURL.path
                 switch type {
                 case "virtiofs", "bind":
-                    fs.source = absPath
+                    // Check if it's an absolute directory path first
```

Comment:
I'd just check if src is a file/dir first, and if that fails assume it's a volume.

Reply:
However, what I'd do here is if you do fall into the "is this a volume" block. I'd query the daemon to see if it actually is a valid volume, and use the returned data to create a .block FSType directly

Reply:
What we COULD do is:
```swift
struct ParsedVolume {
    let name: String
    let destination: String
}

enum FileSystemOrVolume {
   case fs(Filesystem)
   case volume(ParsedVolume)
}
```

And have this method return something like that instead. Then later on it's very simple to tell whether you don't need to do anything, or you need to query for if it's a volume.